bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/DirectoryDialog.java

```diff
@@ -123,9 +123,28 @@ public String getMessage () {
  * </ul>
  */
 public String open() {
+	return show().getSelectedPath();
+}
+
+/**
+ * Makes the dialog visible and brings it to the front of the display.
+ * Equal to {@link DirectoryDialog#open()} but also exposes for state information like user cancellation.
+ *
+ * @return a result object with the absolute path of the selected directory,
+ *         or a state describing that the dialog was cancelled or an error occurred
+ *
+ * @exception SWTException <ul>
+ *    <li>ERROR_WIDGET_DISPOSED - if the dialog has been disposed</li>
+ *    <li>ERROR_THREAD_INVALID_ACCESS - if not called from the thread that created the dialog</li>
+ * </ul>
+ *
+ * @since 3.125
+ */
+public FileDialogResult show() {
```

Comment:
```suggestion
public FileDialogResult selectFile() {
```

with the above suggestion about optional one can use

`dialog.selectFile().getPath().ifPresent(path-> ... do something here if not canceled and not an error ...)`

Reply:
I now changed the name to `askSelectFile()` in bff9b11. To me `selectFile()` suggests that the caller does some selection, but it only asks for selection by the user.